runner.py

```diff
@@ -750,7 +750,11 @@ def setup_services(self):
                 continue  # setup commands are optional
             print('Running commands')
             for cmd in service['setup-commands']:
-                d_command = ['docker', 'exec', container_name, 'sh', '-c', cmd] # This must be a list!
+                if shell := service.get('shell', False):
+                    d_command = ['docker', 'exec', container_name, shell, '-c', cmd] # This must be a list!
+                else:
+                    d_command = ['docker', 'exec', container_name, *cmd.split()] # This must be a list!
```

Comment:
@ArneTR why is `cmd` handled differently here? Shouldn't we either assume a list or always split?

Reply:
Escaping needs to happen on a one by one level if we want to send it directly to docker, otherwise the CLI does not understand the arguments. in the shell case we want the exact opposite, as we do not want to escape the quoted block